src/http/server.clj

```diff
@@ -293,6 +294,29 @@
          (log/infof "\"%s %s HTTP/1.1\" %s %s" method'  uri status len))
        response))))
 
+(defn password-prompt! []
+  (println "Enter password: ")
+  (let [pw (-> (System/console) (.readPassword) (String.))]
```

Comment:
We can remove `println` at all and pass prompt text as a param to `.readPassword`. See docs [here](https://docs.oracle.com/javase/7/docs/api/java/io/Console.html#readPassword(java.lang.String,%20java.lang.Object...)).

Reply:
Another potential problem... `System/console` might return NULL. We need to catch this, print the error and exit with non-zero code. Otherwise, it will print NullPointer stacktrace.

Reply:
Convertion to `String` with no params would force usage of `defaultCharset`... In most cases that would be UTF-8 on modern system, but we can have a problem here. As far as I know there's no generic way how to get charset of the current shell, so I think it's better to stick to UTF-8 here.